apps/frontend/src/components/ResponseBox/ResponseBox.tsx

```diff
@@ -1,28 +1,87 @@
 import styles from './ResponseBox.module.css';
-import { ResponseWithUser } from "../../types"
-import acceptResponse from "../../services/responses";
+import { ResponseWithUserAndRequest } from "../../types"
+import { acceptResponse, deleteResponse } from "../../services/responses";
 import { useRevalidator } from 'react-router';
 import CustomBtn from '../CustomBtn/CustomBtn';
 
 type Props = {
-  response: ResponseWithUser;
+  response: ResponseWithUserAndRequest;
+  requestOwnerId: number;
 }
 
-export default function ResponseBox({ response }: Props) {
+function isLoggedUserRequestOwner(userId: number, requestOwnerId: number) {
+  return userId === requestOwnerId;
+};
+
+function isLoggedUserResponseOwner(userId: number, responseOwnerId: number) {
+  return userId === responseOwnerId;
+}
+
+export default function ResponseBox({ response, requestOwnerId }: Props) {
   const revalidator = useRevalidator();
 
+  let loggedUser = localStorage.getItem('user');
+  let loggedUserId = loggedUser ? JSON.parse(loggedUser).id : null;
+
   const date = String(response.time_created).split("T")[0];
+
   function handleAcceptOffer() {
     acceptResponse(String(response.id));
     revalidator.revalidate();
```

Comment:
In order to avoid using the revalidator, we could do something like what I did for the join neighbourhood button. If you merge main into your branch, you should be able to expand from that in the same vein (i.e. creating a form for each button with a hidden input with `intent` and make the appropriate request based on that value). I'm only saying this because i feel the use of revalidator is discouraged when there's an alternative (from the docs: If you find yourself using this for normal CRUD operations on your data in response to user interactions, you're probably not taking advantage of the other APIs like `<Form>`, useSubmit or useFetcher that do this automatically.).

Reply:
My reason for using re-validator was to trigger a data re-load so that the response box is re-rendered on the spot. I'm not sure if what you suggest will help with that? Maybe we can meet up discuss this

Reply:
messaged you on discord